### ISSUE TITLE
Comments (and some other statements) break succeeding match rules

### ISSUE THREAD

**Issue by FireyFly** [delphinus/vim-firestore#8]
First of all, thanks for writing this plugin/the syntax file! Really handy when you get thrown into a new environment. :)

I noticed whilst writing my Firestore rules that it seems there's a problem with how some statements (as well as comments) are defined currently.  Try highlighting the following sample file:
```
rules_version = '2';

service cloud.firestore {
  match /databases/{database}/documents {
    // Foo
    match /foo/{document=**} { allow read: if true; }
    match /bar/{document=**} { allow read: if true; }

    // Baz
    match /baz/{document=**} {
      allow read, write: if true;
    }
  }
}
```

You'll notice that `service` is marked as unrecognised, ditto with `allow`, `read`, `write`, `if`, `true` inside the "baz" match block, and the colouring after the comments looks a bit funny.  In fact, this appears to trigger with the sample firebase file in the repo as well.

I believe this is due to the ordering of the `nextgroup=` in some of the highlighting rules.  For instance, `firestoreComment` specifies `nextgroup=@firestoreValue,...`, so I believe the next token mistakenly gets highlighted as an identifier.

I think it might be easier to rely a bit less on `nextgroup`, and e.g. for comments I think it makes sense to not specify anything in particular and just let the highlighter match things as on the top-level.  That said, it's been a fair while since I last wrote vim syntax files...

Anyway, I figured I should be a nice citizen and report this, even if I'll just work around the issue locally in the meantime. :)

**Comment by delphinus** [comment]
Thx for reports and deeper inspection! I think this is because the syntax lacks logic for the token: `rules_version`. I have thought `service` was always the top of files...it was wrong.

Soon I will fix this. 👍

**Comment by delphinus** [comment]
a quick workaround here => #9 

I will clean up sources and more debug.

**Comment by hotchpotch** [comment]
@delphinus 
It works well !
 If you can merge please 🙏

**Comment by delphinus** [comment]
@hotchpotch thx for report! I merged #9.
